packages/checkout/widgets-lib/src/widgets/sale/functions/getTopUpViewData.ts

```diff
@@ -0,0 +1,55 @@
+import { ItemType, TransactionRequirement } from '@imtbl/checkout-sdk';
+
+const tokenInfo = (req?: TransactionRequirement) => {
+  if (req && req.current.type !== ItemType.ERC721) {
+    return req.current.token;
+  }
+  return undefined;
+};
+
+export const getTopUpViewData = (
+  transactionRequirements: TransactionRequirement[],
+) => {
+  const native = transactionRequirements.find(
+    ({ type }) => type === ItemType.NATIVE,
+  );
+  const erc20 = transactionRequirements.find(
+    ({ type }) => type === ItemType.ERC20,
+  );
+
+  const balances = {
+    erc20: {
+      value: erc20?.delta.formattedBalance,
+      symbol: tokenInfo(erc20)?.symbol,
+    },
+    native: {
+      value: native?.delta.formattedBalance,
```

Comment:
Is it worth defaulting this value to zero if native not found in transactionRequirements? .. or doesn't matter so much as symbol won't be defined anyway?
```suggestion
      value: native?.delta.formattedBalance ?? 0,
```